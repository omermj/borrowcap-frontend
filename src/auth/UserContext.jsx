import React from "react";

/** Context to provide currentUser thoughout the app */

const UserContext = React.createContext();

export default UserContext;
