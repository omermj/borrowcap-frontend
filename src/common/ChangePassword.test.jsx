import { render } from "@testing-library/react";
import ChangePasswordForm from "./ChangePassword";
import { BrowserRouter } from "react-router-dom";
import UserContext from "../auth/UserContext";

// smoke test
it("renders without crashing", () => {
  render(
    <BrowserRouter>
      <UserContext.Provider value={{ currentUser: { username: "test" } }}>
        <ChangePasswordForm />
      </UserContext.Provider>
    </BrowserRouter>
  );
});

// snapshot test
it("matches snapshot", () => {
  const { asFragment } = render(
    <BrowserRouter>
      <UserContext.Provider value={{ currentUser: { username: "test" } }}>
        <ChangePasswordForm />
      </UserContext.Provider>
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
