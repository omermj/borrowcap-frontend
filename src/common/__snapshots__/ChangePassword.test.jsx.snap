// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="form-wrapper"
  >
    <div
      class="form-inner"
    >
      <h3
        class="text-center mb-4"
      >
        Change Password
      </h3>
      <form
        class=""
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="oldPassword"
          >
            Current Password
          </label>
          <input
            class="form-control"
            id="oldPassword"
            name="currentPassword"
            placeholder="Enter current password"
            required=""
            type="password"
            value=""
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="newPassword"
          >
            New Password
          </label>
          <input
            class="form-control"
            id="newPassword"
            name="newPassword"
            placeholder="Enter new password"
            required=""
            type="password"
            value=""
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="newPasswordRetype"
          >
            Re-type Password
          </label>
          <input
            class="form-control"
            id="newPasswordRetype"
            name="newPasswordRetype"
            placeholder="Re-type password"
            required=""
            type="password"
            value=""
          />
        </div>
        <div
          class="d-grid"
        >
          <button
            class="mt-4 btn-dark btn btn-primary"
            type="submit"
          >
            Update
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
